Task/swaggerize remaining endpoints and folder cleanup

Jsyro opened the pull request:
almost Finish Swagger refactor.

I have excluded mine_party_appointments, hoping to get that in this sprint, but it may have to be dealt with as it's own item.

thegentlemanphysicist commented:
Application status codes seem to be having difficulties being fetched 
![image](https://user-images.githubusercontent.com/9705602/65724562-850c3800-e065-11e9-86e2-bf25fcb9dcab.png)

thegentlemanphysicist commented:
Cannot create a mine on this openshift pod

NWCalvank commented on python-backend/app/api/mines/namespace.py:
```diff
@@ -61,6 +60,10 @@
 
+api.add_resource(ApplicationListResource, '/<string:mine_guid>/applications')
+api.add_resource(ApplicationResource, '/<string:mine_guid>/applications/<string:application_guid>')
+api.add_resource(ApplicationStatusCodeResource, '/applications/status-codes')
```
Along with what Jon mentioned, I'm concerned about the API endpoint changes here. Since I don't see the accompanying changes in the FE

Jsyro replied:
Now they have been added and Paul has identified the constants.

Jsyro commented:
Went into a bit of folder/structure clean up. 

Tests folders match the app folder closer, 
removed namespace folders, 
made them all use 'response_models.py' 
pulled out compliance article as it has nothing to do with a mine.
added missing foreign key and table
removed python date defaults
all endpoints returning our data uses swagger models. 
Please look here thinking from a brand new developer (or someone who wants to call our API and see if it makes sense). https://minesdigitalservices-dev.pathfinder.gov.bc.ca/1025/api/

pdittaro commented on python-backend/app/api/mines/status/resources/status.py:
```diff
@@ -1,28 +1,15 @@
-from ..models.mine_status import MineStatus
 from app.api.mines.status.models.mine_status_xref import MineStatusXref
 from app.extensions import api
 from ....utils.access_decorators import requires_role_view_all
```
Should these Mixins and Decorators be absolute as well?

Jsyro replied:
I'm going to make another PR targeting the destruction of the error mixin.

nayr974 commented on python-backend/app/api/now_submissions/namespace.py:
```diff
@@ -1,8 +1,8 @@
-from app.api.now_submissions.resources.application_resource import ApplicationResource
-from app.api.now_submissions.resources.application_document_resource import ApplicationDocumentResource, ApplicationDocumentTokenResource
-from app.api.now_submissions.resources.application_list_resource import ApplicationListResource
+from .resources.application_resource import ApplicationResource
```
Let's remove, not add relative paths.

nayr974 commented on python-backend/app/api/search/namespace.py:
```diff
@@ -0,0 +1,10 @@
+from flask_restplus import Namespace
+
+from .search.resources.search import SearchResource, SearchOptionsResource
```
Let's remove, not add relative paths.

nayr974 commented on python-backend/app/api/users/minespace/models/minespace_user.py:
```diff
@@ -41,7 +33,7 @@ def find_by_guid(cls, user_guid):
     @classmethod
     def find_by_email(cls, email):
-        return cls.query.filter_by(email=email).filter_by(deleted_ind=False).first()
+        return cls.query.filter_by(email=email).filter_by(deleted_ind=False).all()
```
Are you sure the consuming code expects a collection, and not only one object?

Jsyro replied:
I made that change to simplify the resource, but this is bad because it's a unique column.  i'll change the resource

nayr974 commented on python-backend/app/api/users/namespace.py:
```diff
@@ -0,0 +1,15 @@
+from flask_restplus import Namespace
+
+from .minespace.resources.minespace_user import MinespaceUserResource, MinespaceUserListResource
```
Let's remove, not add relative paths please.

Jsyro commented on python-backend/app/api/compliance/response_models.py:
```diff
@@ -0,0 +1,16 @@
+
+COMPLIANCE_ARTICLE_MODEL = api.model(
+    'ComplianceArticle', {
+        'compliance_article_id': fields.Integer,
```
No there is not. only integer id and then lookup via section/subsection. given it's a representation of a formal public document. I don't think this it huge issue, but yes it looks odd.  My change was only to make a compliance namespace for this data.

Jsyro commented on python-backend/app/api/mines/compliance/resources/compliance.py:
```diff
@@ -12,17 +12,7 @@
-class Date(fields.Raw):
-    def format(self, value):
-        return value.strftime("%Y-%m-%d") if value else None
+from ...mine.models.mine import Mine
```
I've met you in the middle, if it's with in the namespace, i think relative imports make sense, however I went and changed all relative imports of the utils folder to use app.api.utils.

Jsyro commented on python-backend/app/api/mines/mine/resources/mine.py:
```diff
@@ -22,7 +22,7 @@
-from app.api.mines.mine_api_models import MINE_LIST_MODEL, MINE_MODEL
+from app.api.mines.response_models import MINE_LIST_MODEL, MINE_MODEL
 
 # FIXME: Model import from outside of its namespace
```
removed.

pdittaro commented:
Understood from this morning's conversation that we would be removing the relative / namespace imports for the time being before bringing this in?

Jsyro commented:
Reverted response model imports to be absolute. namespace.py files, where the purpose is to register resources further down the folder structure, and don't traverse up the tree with `....`, I think, are easier to manage with the changes I've made. Happy for more discussion though.
